src/lib/peers/PeerInfo.ts

```diff
@@ -1,83 +1,120 @@
 import { AddressInfo } from 'net';
+import { peer } from '../../../proto/js/Peer';
+import { protobufToString, stringToProtobuf } from '../utils';
+const { PeerInfoMessage } = peer;
 
 class Address {
```

Comment:
I don't think this needs to be a class. Use a normal object and TS type. Then the other functions are just utility functions. I only really use classes to marry ops with data that is mutable.

Prefer immutable data most of the time, and if you have immutable data, simple data structures are better with some transform utility functions in the same module is fine.

Reply:
well there are some convenience methods on this class like `parse` but I think you can do the same with TS types and utils.